system/lib/wasmfs/wasmfs.h

```diff
@@ -24,6 +24,7 @@ class WasmFS {
 
   std::vector<std::unique_ptr<Backend>> backendTable;
   FileTable fileTable;
+  backend_t rootBackend;
```

Comment:
I'm not clear on this part. This duplicates state and it might get out of sync with `getRootDirectory()->getBackend()`?

Reply:
Oh yeah, I don't think I considered `getRootDirectory()->getBackend()` as a good way to get the root backend. I'll remove this.